source/_UIAHandler.py

```diff
@@ -353,22 +353,39 @@ def _isUIAWindowHelper(self,hwnd):
 		# There are certain window classes that just had bad UIA implementations
 		if windowClass in badUIAWindowClassNames:
 			return False
-		if windowClass=="NetUIHWND":
-			parentHwnd=winUser.getAncestor(hwnd,winUser.GA_ROOT)
-			# #2816: Outlook 2010 auto complete does not fire enough UIA events, IAccessible is better.
-			# #4056: Combo boxes in Office 2010 Options dialogs don't expose a name via UIA, but do via MSAA.
-			if winUser.getClassName(parentHwnd) in {"Net UI Tool Window","NUIDialog"}:
-				return False
 		# allow the appModule for the window to also choose if this window is bad
 		if appModule and appModule.isBadUIAWindow(hwnd):
 			return False
 		# Ask the window if it supports UIA natively
 		res=windll.UIAutomationCore.UiaHasServerSideProvider(hwnd)
 		if res:
 			# the window does support UIA natively, but
-			# Microsoft Word should not use UIA unless we can't inject or the user explicitly chose to use UIA with Microsoft word
-			if windowClass=="_WwG" and not (config.conf['UIA']['useInMSWordWhenAvailable'] or not appModule.helperLocalBindingHandle):
-				return False
+			# MS Word documents now have a very usable UI Automation implementation. However,
+			# Builds of MS Office 2016 before build 9000 or so had bugs which we cannot work around.
+			# Therefore refuse to use UIA for builds earlier than this, if we can inject in-process.
+			if (
+				# An MS Word document window 
+				windowClass=="_WwG" 
+				# Disabling is only useful if we can inject in-process (and use our older code)
+				and appModule.helperLocalBindingHandle 
+				# Allow the user to explisitly force UIA support for MS Word documents no matter the Office version 
+				and not config.conf['UIA']['useInMSWordWhenAvailable']
+			):
+				# We can only safely check the version of known Office apps using the Word document control.
+				# Other uses for now we just need to assume the implementation is good.
+				if appModule.appName not in ('outlook','winword','excel'):
+					log.debugWarning("Unknown application using MS Word document control: %s"%appModule.appName)
+					return True
+				try:
+					versionMajor,versionMinor,versionBuild,versionPatch=[int(x) for x in appModule.productVersion.split('.')]
+				except Exception as e:
+					log.error("Error parsing versioninformation %s, %s"%(appModule.productVersion,e))
+					return True
+				if (
+					versionMajor<16
+					or versionMajor==16 and versionMinor==0 and versionBuild<9000

```

Comment:
I think I would prefer to see these values (16, 0, 9000) put in a named constant.